config_converter/config_mapper/config_mapper.py

```diff
@@ -0,0 +1,198 @@
+        if p.name not in outer_map:
+            if p.name in _UNSUPPORTED_FIELDS:
+                print(f'parameter {p.name} cant be converted')
+            elif p.name in special_params:
+                _map_special_fields(specific, p, plugin)
+            elif p.name not in main_map:
+                print(f'parameter {p.name} is unknown')
+            else:
+                specific[main_map[p.name]] = p.value
+    for nd in d.directives:
+        if nd.name not in supported_dirs:
+            print(f'directive {nd.name} is not supported')
+        else:
+            for np in nd.params:
+                _map_special_fields(specific, np, plugin)
+    cur[f'{cur["type"]}_{d.name}_config'] = specific
+    return cur
+
+
+def _map_special_fields(specific: dict, p: config_pb2.Param,
+                        plugin: str) -> None:
+    """Cases on p with all special_map dict, calls a map function."""
+    if p.name in _SPECIAL_MAP[plugin]:
+        if plugin == 'in_tail':
+            _map_parser_dir(specific, p)
+
+
+def _map_parser_dir(specific: dict, p: config_pb2.Param) -> None:
+    """Create parser dir in master config."""
+    parser_type_map = {
+        'multiline': 'multiline',
+        'none': 'none',
+        'regex': 'regex',
+        'apache2': 'regex',
+        'apache_error': 'regex',
+        'json': 'json',
+        'nginx': 'regex'
+    }
+    specific['parser'] = specific.get('parser', {})
+    if p.name == 'expression':
+        specific['parser']['regex_parser_config'] = \
+                specific['parser'].get('regex_parser_config', {})
+        specific['parser']['regex_parser_config'][p.name] = p.value
+    elif p.name == 'format' or p.name == '@type':
+        if p.value not in parser_type_map:
+            print('unknown parser format type ' + p.value)
+        else:
+            specific['parser']['type'] = parser_type_map[p.value]
+    else:
+        specific['parser']['multiline_parser_config'] = \
+                specific['parser'].get('multiline_parser_config', {})
+        if p.name in [f'format{i}' for i in range(1, 21)]:
+            format_number = p.name[6:]
+            specific['parser']['multiline_parser_config']\
+                    [f'format_{format_number}'] = p.value
+        elif p.name == 'multiline_flush_interval':
+            specific['parser']['multiline_parser_config']['flush_interval']\
+                    = p.value
+        else:
+            specific['parser']['multiline_parser_config'][p.name] = p.value
+
+
+def write_to_yaml(result: dict, path: str, name: str) -> None:
+    """Writes created result dictionary to a yaml file."""
+    with open(f'{path}/{name}.yaml', 'w') as f:
+        yaml.dump(result, f)
+
+
+def read_file(path: str) -> str:
+    """Returns contents of file at given path."""
+    with open(path, 'rt') as f:
+        return f.read()
+
+
+if __name__ == '__main__':
+    master_path, file_name = sys.argv[1], sys.argv[2]
+    config_json = sys.argv[-1]
```

Comment:
Could this be

    master_path, file_name, config_json = sys.argv[1], sys.argv[2], sys.argv[3]

?

Reply:
Yep will change!